packages/loader/container-loader/src/container.ts

```diff
@@ -244,30 +245,35 @@ export class CollabWindowTracker {
         }
 
         // We don't acknowledge no-ops to avoid acknowledgement cycles (i.e. ack the MSN
-        // update, which updates the MSN, then ack the update, etc...).
-        if (message.type === MessageType.NoOp) {
+        // update, which updates the MSN, then ack the update, etc...). Also, don't
+        // count system messages in ops count.
+        if (message.type === MessageType.NoOp || isSystemMessage(message)) {
```

Comment:
MessageType.NoOp is a system message, so we can exclude it from check here / add assert after it